### PR TITLE
Add Adjust device mode integration sample & example app

### PR DESCRIPTION
**Pull request by bsneed**

### CODE REVIEW

**Review by alanjcharles — APPROVED** [review]

File: Examples/destination_plugins/AdjustDestination.swift [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    public func update(settings: Settings) {
+        guard let settings: AdjustSettings = settings.integrationSettings(name: "Adjust") else { return }
+        self.settings = settings
+        
+        var environment = ADJEnvironmentSandbox
+        if let _ = settings.setEnvironmentProduction {
+            environment = ADJEnvironmentProduction
+        }
+        
+        let adjustConfig = ADJConfig(appToken: settings.appToken, environment: environment)
```
Comment (migs647) [inline Examples/destination_plugins/AdjustDestination.swift:76]: Do we need to protect against an empty appToken? Can it be nil?

File: Examples/destination_plugins/AdjustDestination.swift [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            adjustConfig?.delegate = self
+        }
+        
+        if let useDelay = settings.setDelay, useDelay == true, let delayTime = settings.delayTime {
+            adjustConfig?.delayStart = delayTime
+        }
+        
+        Adjust.appDidLaunch(adjustConfig)
+
+        started = true
```
Comment (migs647) [inline Examples/destination_plugins/AdjustDestination.swift:92]: Do we want to queue up any of these events if started hasn't happened yet?

.. If so, should we consider adding that feature to the core library?

**Review by migs647 — APPROVED** [review]
LGTM

File: Examples/destination_plugins/AdjustDestination.swift (hunk 1 above)
Comment (bsneed) [reply Examples/destination_plugins/AdjustDestination.swift:76]: It potentially could be, but let adjust report that error.

File: Examples/destination_plugins/AdjustDestination.swift (hunk 2 above)
Comment (bsneed) [reply Examples/destination_plugins/AdjustDestination.swift:92]: Possibly; working on that separately from this PR.

**Pull request merged**